router/network_router.go

```diff
@@ -204,3 +206,41 @@ func (router *NetworkRouter) relayBroadcast(srcPeer *mesh.Peer, key PacketKey) F
 
 	return op
 }
+
+// Persisting the set of peers we are supposed to connect to
+const peersIdent = "directPeers"
+
+func (router *NetworkRouter) persistPeers() {
+	if err := router.db.Save(peersIdent, router.ConnectionMaker.Targets()); err != nil {
+		log.Errorf("Error persisting peers: %s", err)
+		return
+	}
+}
+
+func (router *NetworkRouter) SetInitialPeers(peers []string) ([]string, []error) {
+	var storedPeers []string
+	if err := router.db.Load(peersIdent, &storedPeers); err != nil {
+		return nil, []error{err}
+	}
+
+	if !equal(peers, storedPeers) {
+		log.Println("Overriding command-line peer list with stored list:", peers)
```

Comment:
You're missing the assignment of storedPeers to peers in the event they are unequal